cpp/include/cugraph_c/graph.h

```diff
@@ -117,18 +172,50 @@ cugraph_error_code_t cugraph_sg_graph_create_from_csr(
   cugraph_error_t** error);
 
 /**
- * @brief     Destroy an SG graph
+ * @brief     Construct an SG graph from a CSR input
  *
- * @param [in]  graph  A pointer to the graph object to destroy
+ * @param [in]  handle         Handle for accessing resources
+ * @param [in]  properties     Properties of the constructed graph
+ * @param [in]  offsets        Device array containing the CSR offsets array
+ * @param [in]  indices        Device array containing the destination vertex ids
+ * @param [in]  weights        Device array containing the edge weights.  Note that an unweighted
+ *                             graph can be created by passing weights == NULL.
+ * @param [in]  edge_ids       Device array containing the edge ids for each edge.  Optional
+                               argument that can be NULL if edge ids are not used.
+ * @param [in]  edge_type_ids  Device array containing the edge types for each edge.  Optional
+                               argument that can be NULL if edge types are not used.
+ * @param [in]  store_transposed If true create the graph initially in transposed format
+ * @param [in]  renumber       If true, renumber vertices to make an efficient data structure.
+ *    If false, do not renumber.  Renumbering enables some significant optimizations within
+ *    the graph primitives library, so it is strongly encouraged.  Renumbering is required if
+ *    the vertices are not sequential integer values from 0 to num_vertices.
+ * @param [in]  do_expensive_check    If true, do expensive checks to validate the input data
+ *    is consistent with software assumptions.  If false bypass these checks.
+ * @param [out] graph          A pointer to the graph object
+ * @param [out] error          Pointer to an error object storing details of any error.  Will
+ *                             be populated if error code is not CUGRAPH_SUCCESS
+ *
+ * @return error code
  */
-// FIXME:  This should probably just be cugraph_graph_free
-//         but didn't want to confuse with original cugraph_free_graph
-void cugraph_sg_graph_free(cugraph_graph_t* graph);
+cugraph_error_code_t cugraph_graph_create_sg_from_csr(
```

Comment:
Since `cugraph_graph_create_sg` supports flags to drop both self loops and multi edges through the methods `cugraph::remove_self_loops` and `cugraph::sort_and_remove_multi_edges`, shouldn't `cugraph_graph_create_sg_from_csr` also support those flags? In fact, `cugraph_graph_create_sg_from_csr` extracts the `COO` from the `CSR` before calling `cugraph::create_graph_from_edgelist` therefore, we can drop self loops and multi edges (if set to True) right before the graph question.

Reply:
We could add that feature.  I was assuming that if you already have a CSR as input then you have already done whatever ETL steps are desired.